src/textual/widgets/_data_table.py

```diff
@@ -138,6 +157,53 @@ def __rich_repr__(self):
         yield "column_key", self.column_key
 
 
+class KeysToIndices(TwoWayDict[StringKeySubclass, int]):
```

Comment:
What is the purpose of this class over a plain `TwoWayDict`? Is it to generate a more specific exception? Was there a specific case were the DataTable was generating a non specific exception?

Reply:
`TwoWayDict` was lenient and allowed returning `None`s from its `get` methods.
When that was correctly typed, many typing errors appeared because the code does not contemplate the possibility of getting `None`s back from its internal bidirectional dictionaries.

I started tackling those one by one, doing a runtime check for `None` and raising the appropriate error.

As I did that, I realised the code was getting bloated with all those checks.
So, I decided to lift those checks to the specialised bidirectional dictionary `KeysToIndices` that is responsible for raising the appropriate errors when fetching a key or a value fails.
(Which the semantics of the data table code imply should never happen.)

I thought this was a good solution but I can be convinced otherwise.

If there are no plans to use the general bidirectional dictionary elsewhere, we can merge `TwoWayDict` and `KeysToIndices` into a single class.

Reply:
I have a suspicion that the DataTable code checks for these conditions, so the TWD never returns None at runtime.

Perhaps the `get` and `get_key` should just use `[]` syntax.

@darrenburns Can you confirm?

Reply:
I just merged main and I have a couple of failing tests because I'm getting `None`s from `get`/`get_key`.